src/main/java/net/codejava/MvcController.java

```diff
@@ -207,11 +213,16 @@ public String submitWithdraw(@ModelAttribute("user") User user) {
       List<Map<String,Object>> queryResults = jdbcTemplate.queryForList(getUserNameAndBalanceSql);
       Map<String,Object> userData = queryResults.get(0);
 
-      user.setFirstName((String)userData.get("FirstName"));
-      user.setLastName((String)userData.get("LastName"));
-      user.setBalance((int)userData.get("Balance"));
+      int amount = user.getAmountToWithdraw();
+      // check if withdraw input is valid 
+      if (amount >= 0) {

```

Comment:
It might be more readable to approach your negative input checks in the opposite way. You can just check whether `user.getAmountToWithdraw()` is negative at the very beginning of the method, and immediately return `"welcome"` to skip all the method logic (like the `UPDATE` and `SELECT` SQL commands.